doc/source/user_guide/categorical.rst

```diff
@@ -813,16 +813,16 @@ but the categories of these categoricals need to be the same:
     res
     res.dtypes
 
-In this case the categories are not the same, and therefore an error is raised:
+If the categories are not exactly the same, merging will coerce the
```

Comment:
I agree it could be more clear, but I'm struggling with trying to keep the terminology consistent with existent docs. How about this?

> Otherwise the result's dtype will be determined by promotion of the mergands' categories' dtypes.

Reply:
How about

> If the categories are not identical, categorical columns will be cast to a regular array with
> the dtype of the underlying categories, which will likely have higher memory usage.

Reply:
That sounds good. Should there be an example which doesn't cast to object to make that clear? Something like this:

```python
In [101]: pd.concat([ 
     ...:     pd.Series([1, 2], dtype="category"), 
     ...:     pd.Series([3., 4.], dtype="category") 
     ...: ])                                                                                        
Out[101]: 
0    1.0
1    2.0
0    3.0
1    4.0
dtype: float64
```